### PR TITLE
dealii, piranha, trilinos: remove mentions of non-existing variants

### PR DESCRIPTION
**Pull request by alalazo**

### CODE REVIEW

File: var/spack/repos/builtin/packages/trilinos/package.py [hunk 1]
```diff
@@ -398,7 +396,6 @@ class Trilinos(CMakePackage, CudaPackage):
 
     # Dependencies/conflicts when MPI is disabled
     depends_on('hdf5~mpi', when='+hdf5~mpi')
-    conflicts('+parmetis', when='~mpi')
```
Comment (haampie) [inline var/spack/repos/builtin/packages/trilinos/package.py:401]: Maybe this was supposed to be `+metis`? But then again, removing this doesn't change the current behaviour, so I'm fine with it.

**Review by haampie — APPROVED** [review]

File: var/spack/repos/builtin/packages/trilinos/package.py (hunk 1 above)
Comment (sethrj) [reply var/spack/repos/builtin/packages/trilinos/package.py:401]: I think the updated behavior is correct.

**Review by sethrj — COMMENTED** [review]
Trilinos changes look good, thanks!

File: var/spack/repos/builtin/packages/dealii/package.py [hunk 2]
```diff
@@ -219,7 +219,6 @@ class Dealii(CMakePackage, CudaPackage):
           when='@9.0.0:9.1.1')
 
     # Check for sufficiently modern versions
-    conflicts('cxxstd=98', when='@9.0:')
```
Comment (jppelteret) [inline var/spack/repos/builtin/packages/dealii/package.py:222]: Sorry that I didn't see your review request until now @alalazo. Thanks a lot for the fix!

**Pull request merged**